utils/backup/S3/fo-restore-s3

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+# Copyright (C) Siemens AG 2017
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# version 2 as published by the Free Software Foundation.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+source ${DIR}/before
+
+set -x
+
+echo Starting restore at `date`
+
+#### stop fossology scheduler
+echo "Stopping fossology scheduler..."
+/etc/init.d/fossology stop
+
+#### stop apache2
+echo "Stopping apache2..."
+/etc/init.d/apache2 stop
+
+echo Start removing old data at `date`
+
+${DIR}/../../fo-cleanold --delete-repository --delete-database
+
+#### sync full tar
+echo Done removing old data. Starting download+extracting of ${FILENAME} at `date`
+${DIR}/venv/bin/aws ${ENDPOINT} --debug s3 cp --expected-size 200000000000 --sse AES256 s3://${BUCKET}/${FILENAME} - | tar --warning=no-timestamp -xzf - -C /
+
+echo Checking diff of repository at `date`
+DIFF=`${DIR}/venv/bin/aws s3 cp --sse AES256 s3://${BUCKET}/${FILENAME}-repo-diff.txt -`
+CURRENT_DIFF=`ls -RlahGg --time-style='+' /srv/fossology/repository/ | cut -d ' ' -f 2-5`
```

Comment:
You are calling `ls -RlahGg --time-style='+' /srv/fossology/repository/` at multiple places, please move them to an function.